system-test/testnet-performance/testnet-automation.sh

```diff
@@ -0,0 +1,187 @@
+[[ -n $TESTNET_TAG ]] || TESTNET_TAG=testnet-automation
+[[ -n $INFLUX_HOST ]] || INFLUX_HOST=https://metrics.solana.com:8086
+
+# TODO: Remove all default values, force explicitness in the testcase definition
+[[ -n $TEST_DURATION ]] || TEST_DURATION=300
+[[ -n $RAMP_UP_TIME ]] || RAMP_UP_TIME=60
+[[ -n $NUMBER_OF_VALIDATOR_NODES ]] || NUMBER_OF_VALIDATOR_NODES=2
+[[ -n $NUMBER_OF_CLIENT_NODES ]] || NUMBER_OF_CLIENT_NODES=1
+[[ -n $TESTNET_ZONES ]] || TESTNET_ZONES="us-west1-a"
+
+function collect_logs {
+  echo --- collect logs from remote nodes
+  rm -rf net/log
+  net/net.sh logs
+  # shellcheck disable=SC2045
+  for logfile in $(ls -A net/log) ; do
+    (
+      cd net/log
+      new_log="$TESTNET_TAG"_"$NUMBER_OF_VALIDATOR_NODES"-nodes_"$(basename "$logfile")"
+      cp "$logfile" "$new_log"
+      upload-ci-artifact "$new_log"
+    )
+  done
+}
+
+function cleanup_testnet {
+  (
+    set +e
+    collect_logs
+  )
+
+  (
+    set +e
+    echo --- Stop Network Software
+    net/net.sh stop
+  )
+
+  case $CLOUD_PROVIDER in
+  gce)
+  (
+    cat <<EOF
+- wait: ~
+  continue_on_failure: true
+
+- command: "net/gce.sh delete -p ${TESTNET_TAG}"
+  label: "Delete Testnet"
+  agents:
+    - "queue=testnet-deploy"
+EOF
+  ) | buildkite-agent pipeline upload
+  ;;
+  colo)
+    (
+    cat <<EOF
+- wait: ~
+  continue_on_failure: true
+
+- command: "net/colo.sh delete -p ${TESTNET_TAG}"
+  label: "Delete Testnet"
+  agents:
+    - "queue=colo-deploy"
+EOF
+  ) | buildkite-agent pipeline upload
+  ;;
+  *)
+    echo "Error: Unsupported cloud provider: $CLOUD_PROVIDER"
+    ;;
+  esac
+}
+trap cleanup_testnet EXIT
+
+cd "$(dirname "$0")/../.."
+
+if [[ -z $CHANNEL ]]; then
+  echo --- downloading tar from build artifacts
+  buildkite-agent artifact download "solana-release*.tar.bz2" .
+fi
+
+# shellcheck disable=SC1091
+source ci/upload-ci-artifact.sh
+
+if [[ -z $SOLANA_METRICS_CONFIG ]]; then
+  if [[ -z $SOLANA_METRICS_PARTIAL_CONFIG ]]; then
+    echo SOLANA_METRICS_PARTIAL_CONFIG not defined
+    exit 1
+  fi
+  export SOLANA_METRICS_CONFIG="db=$TESTNET_TAG,host=$INFLUX_HOST,$SOLANA_METRICS_PARTIAL_CONFIG"
+fi
+echo "SOLANA_METRICS_CONFIG: $SOLANA_METRICS_CONFIG"
+
+maybeClientOptions=
+if [[ -n $CLIENT_OPTIONS ]] ; then
+  maybeClientOptions="-c"
+fi
+
+maybeMachineType=
+if [[ -n $VALIDATOR_NODE_MACHINE_TYPE ]] ; then
+  maybeMachineType="-G"
+fi
+
+TESTNET_CLOUD_ZONES=(); while read -r -d, ; do TESTNET_CLOUD_ZONES+=( "$REPLY" ); done <<< "${TESTNET_ZONES},"
+
+launchTestnet() {
+  set -x
+
+  # shellcheck disable=SC2068
+  echo --- create "$NUMBER_OF_VALIDATOR_NODES" nodes
+
+  case $CLOUD_PROVIDER in
+    gce)
+    # shellcheck disable=SC2068
+      net/gce.sh create \
+        -d pd-ssd \
+        -n "$NUMBER_OF_VALIDATOR_NODES" -c "$NUMBER_OF_CLIENT_NODES" \
+        "$maybeMachineType" "$VALIDATOR_NODE_MACHINE_TYPE" \
+        -p "$TESTNET_TAG" ${TESTNET_CLOUD_ZONES[@]/#/-z } "$ADDITIONAL_FLAGS"
```

Comment:
It might pay to quote `-z ` to make clear that the trailing space is intentional